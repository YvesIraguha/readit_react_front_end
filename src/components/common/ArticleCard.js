import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";
import { withRouter } from "react-router-dom";
export class ArticleCard extends Component {
  constructor(props) {
    super(props);
    const { id, title, content } = this.props;
    this.state = {
      id,
      title,
      content
    };
  }
  render() {
    const { id, title, content } = this.state;
    const {
      history: { push }
    } = this.props;

    return (
      <div className="main-article" onClick={() => push(`/articles/${id}`)}>
        <h1>{title}</h1>
        {ReactHtmlParser(content)}
      </div>
    );
  }
}

ArticleCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  history: PropTypes.shape({}).isRequired
};
export default withRouter(ArticleCard);
