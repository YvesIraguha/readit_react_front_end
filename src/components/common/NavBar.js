import React from "react";
import openBook from "../../assets/icons/open-book.svg";
import userAvatar from "../../assets/images/user.png";
// eslint-disable-next-line react/display-name
export default () => (
  <section className="navbar">
    <div className="logo">
      <img src={openBook} alt="Logo" className="icon" />
    </div>
    <div className="profile">
      <div className="avatar">
        <img src={userAvatar} alt="Profile" className="icon" />
      </div>
      <div className="username">
        <p>Joe Doe</p>
      </div>
    </div>
  </section>
);
