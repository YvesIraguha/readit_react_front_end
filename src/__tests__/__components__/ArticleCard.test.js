import React from "react";
import { shallow } from "enzyme";
import { ArticleCard } from "../../components/common/ArticleCard";

const props = {
  history: {
    push: jest.fn()
  },
  id: "helloWorld",
  title: "hello world",
  content: "hello world"
};
describe("article card", () => {
  test("should render article card correctly", () => {
    const wrapper = shallow(<ArticleCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should redirect to the article route on click", () => {
    const wrapper = shallow(<ArticleCard {...props} />);
    wrapper.find(".main-article").simulate("click");
    expect(props.history.push).toHaveBeenCalledWith("/articles/helloWorld");
  });
});
