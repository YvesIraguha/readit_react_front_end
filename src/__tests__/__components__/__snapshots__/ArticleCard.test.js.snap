// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article card should render article card correctly 1`] = `
<div
  className="main-article"
  onClick={[Function]}
>
  <h1>
    hello world
  </h1>
  hello world
</div>
`;
